### PR TITLE
Adding travis yml file with sonarcloud

### PR DESCRIPTION
**Pull request by morgantaschuk**

### CODE REVIEW

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 1]
```diff
@@ -462,4 +458,5 @@ private String makeBasename(final String name) {
         }
         return basename;
     }
+    //FIXME
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)

**Comment by sqwci** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L71](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L71): Refactor this method to reduce its Cognitive Complexity from 37 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L74](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L74): Define a constant instead of duplicating this literal "input_files" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L229](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L229): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L257](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L257): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L19](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L19): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L171](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L171): Put single-quotes around '_' to use the faster "indexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L177](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L177): Put single-quotes around '/' to use the faster "lastIndexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L179](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L179): Put single-quotes around '.' to use the faster "indexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L272](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L272): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L303](https://github.com/oicr-gsi/sample-fingerprinting/blob/1f0c43e388ee662c7cd3caf567823a900648aa5b/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L303): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 2]
```diff
@@ -70,16 +59,17 @@
     private String bundledJRE;
     private String[] rgDetails;
     private boolean doBamFix = false;
+    private static final String INPUT_FILES="input_files";
 
     @Override
     public Map<String, SqwFile> setupFiles() {
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java:62]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 35 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 3]
```diff
@@ -70,16 +59,17 @@
...
         // Set up reference, bam and vcf files here
         try {
-            if (getProperty("input_files") == null) {
-                Logger.getLogger(FingerprintCollectorWorkflow.class.getName()).log(Level.SEVERE, "input_files is not set, we need at least one bam file");
+            if (getProperty(INPUT_FILES) == null) {
+                Logger.getLogger(FingerprintCollectorWorkflow.class.getName()).log(Level.SEVERE, INPUT_FILES+" is not set, we need at least one bam file");
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by sqwci** [comment]
SonarQube analysis reported 13 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L223](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L223): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L251](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L251): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L266](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L266): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L384](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L384): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L444](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L444): Put single-quotes around '/' to use the faster "lastIndexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L446](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L446): Put single-quotes around '.' to use the faster "indexOf(char)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)
1. ![MINOR][MINOR] [FingerprintCollectorWorkflow.java#L26](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L26): Remove the "studyName" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)
1. ![MINOR][MINOR] [FingerprintCollectorWorkflow.java#L28](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L28): Make this final field static too. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)
1. ![MINOR][MINOR] [FingerprintCollectorWorkflow.java#L29](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L29): Make this final field static too. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)
1. ![MINOR][MINOR] [FingerprintCollectorWorkflow.java#L30](https://github.com/oicr-gsi/sample-fingerprinting/blob/0c067cb89303a0d25d1b37085cc667998a8c14a6/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L30): Make this final field static too. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 4]
```diff
@@ -34,11 +23,10 @@
     private String[] bamFiles;
     private String[] vcfFiles;
     private String[] gatkDirs;
-    private String studyName = "";
     private String dataDir;
-    private final String tempDir = "tempfiles/";
-    private final String gatkTmp = "temp";
-    private final String finDir  = "finfiles/";
+    private final static String tempDir = "tempfiles/";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: CRITICAL Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 5]
```diff
@@ -34,11 +23,10 @@
...
+    private final static String gatkTmp = "temp";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: CRITICAL Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 6]
```diff
@@ -34,11 +23,10 @@
...
+    private final static String finDir  = "finfiles/";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: CRITICAL Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 7]
```diff
@@ -214,7 +205,7 @@ public void setupDirectory() {
             this.addDirectory(this.dataDir);
             this.addDirectory(this.tempDir);
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MAJOR Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 8]
```diff
@@ -214,7 +205,7 @@ public void setupDirectory() {
...
             this.addDirectory(this.dataDir + this.finDir);
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MAJOR Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 9]
```diff
@@ -34,11 +23,10 @@
...
-    private String studyName = "";
     private String dataDir;
-    private final String tempDir = "tempfiles/";
-    private final String gatkTmp = "temp";
-    private final String finDir  = "finfiles/";
+    private final static String tempDir = "tempfiles/";
+    private final static String gatkTmp = "temp";
+    private final static String finDir  = "finfiles/";
     //private final int batchCount = 100; // Use for job batching, this many jobs
     private String gatkPrefix = "./";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MINOR Remove the "gatkPrefix" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

**Comment by sqwci** [comment]
SonarQube analysis reported 20 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L224](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L224): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L252](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L252): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L213](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L213): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L267](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L267): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L340](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L340): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L362](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L362): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L363](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L363): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L364](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L364): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L367](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L367): Change this instance-reference to a static reference. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L385](https://github.com/oicr-gsi/sample-fingerprinting/blob/894d514597ef778f9169d05fad3310cdf8563533/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L385): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 10]
```diff
@@ -34,11 +23,10 @@
     private String[] bamFiles;
     private String[] vcfFiles;
     private String[] gatkDirs;
-    private String studyName = "";
     private String dataDir;
-    private final String tempDir = "tempfiles/";
-    private final String gatkTmp = "temp";
-    private final String finDir  = "finfiles/";
+    private final static String TEMP_DIR = "tempfiles/";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 11]
```diff
@@ -34,11 +23,10 @@
...
+    private final static String GATK_TMP = "temp";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 12]
```diff
@@ -34,11 +23,10 @@
...
+    private final static String FINDIR  = "finfiles/";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java [hunk 13]
```diff
@@ -34,11 +23,10 @@
...
     //private final int batchCount = 100; // Use for job batching, this many jobs
     private String gatkPrefix = "./";
```
Comment (sqwci) [inline workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java]: MINOR Remove the "gatkPrefix" field and declare it as a local variable in the relevant methods. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)

**Comment by sqwci** [comment]
SonarQube analysis reported 9 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L224](https://github.com/oicr-gsi/sample-fingerprinting/blob/4f7ce229e631c207fc0bc68ed33a19a5278b4f77/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L224): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L252](https://github.com/oicr-gsi/sample-fingerprinting/blob/4f7ce229e631c207fc0bc68ed33a19a5278b4f77/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L252): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L267](https://github.com/oicr-gsi/sample-fingerprinting/blob/4f7ce229e631c207fc0bc68ed33a19a5278b4f77/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L267): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L385](https://github.com/oicr-gsi/sample-fingerprinting/blob/4f7ce229e631c207fc0bc68ed33a19a5278b4f77/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L385): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by morgantaschuk** [comment]
ok to test

File: workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java (hunk 2 above)
Comment (morgantaschuk) [reply workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java:62]: Not going to fix.

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L222](https://github.com/oicr-gsi/sample-fingerprinting/blob/fedc5fdc2fb44eee0dfa3860353c4b89e67dd0eb/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L222): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L250](https://github.com/oicr-gsi/sample-fingerprinting/blob/fedc5fdc2fb44eee0dfa3860353c4b89e67dd0eb/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L250): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L265](https://github.com/oicr-gsi/sample-fingerprinting/blob/fedc5fdc2fb44eee0dfa3860353c4b89e67dd0eb/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L265): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L383](https://github.com/oicr-gsi/sample-fingerprinting/blob/fedc5fdc2fb44eee0dfa3860353c4b89e67dd0eb/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L383): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L222](https://github.com/oicr-gsi/sample-fingerprinting/blob/b7b1668ef81bb380c5a75e53c4e695338a78017c/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L222): Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [FingerprintCollectorWorkflow.java#L250](https://github.com/oicr-gsi/sample-fingerprinting/blob/b7b1668ef81bb380c5a75e53c4e695338a78017c/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L250): Define a constant instead of duplicating this literal " -jar " 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L265](https://github.com/oicr-gsi/sample-fingerprinting/blob/b7b1668ef81bb380c5a75e53c4e695338a78017c/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L265): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [FingerprintCollectorWorkflow.java#L383](https://github.com/oicr-gsi/sample-fingerprinting/blob/b7b1668ef81bb380c5a75e53c4e695338a78017c/workflow-fingerprint-collector/src/main/java/ca/on/oicr/pde/workflows/FingerprintCollectorWorkflow.java#L383): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by pruzanov — APPROVED** [review]

**Pull request merged**